### PR TITLE
demos: Add support for headless XCB

### PR DESCRIPTION
**Pull request by abergmeier**
For headless XCB we cannot create a surface.
Simply ignore XCB if connection could not be established.

### CODE REVIEW

File: demos/vulkaninfo.c [hunk 1]
```diff
@@ -930,14 +930,15 @@ static void AppCreateXcbWindow(struct AppInstance *inst) {
     if (inst->xcb_connection == NULL) {
         printf("XCB failed to connect to the X server.\nExiting ...\n");
```
Comment (chrisforbes) [inline demos/vulkaninfo.c]: Should fix this error message too, now that we're no longer exiting.

File: demos/vulkaninfo.c [hunk 2]
```diff
@@ -930,14 +930,15 @@ static void AppCreateXcbWindow(struct AppInstance *inst) {
...
         fflush(stdout);
-        exit(1);
+        return;
     }
 
     int conn_error = xcb_connection_has_error(inst->xcb_connection);
     if (conn_error) {
         printf("XCB failed to connect to the X server due to error:%d.\nExiting ...\n", conn_error);
```
Comment (chrisforbes) [inline demos/vulkaninfo.c]: ditto

**Comment by abergmeier** [comment]
Rebase on top of https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/pull/2055 and fixed messages.

**Comment by mark-lunarg** [comment]
Pushed to Master -- thanks!

**Pull request closed without merging**